x/dsmr/node_test.go

```diff
@@ -29,6 +32,29 @@ var (
 	_ Verifier[tx] = (*failVerifier)(nil)
 )
 
+const testingDefaultValidityWindowDuration = int64(5)
+
+type testingChainIndex struct {
+	blocks map[ids.ID]validitywindow.ExecutionBlock[*ChunkCertificate]
+}
+
+func (ti *testingChainIndex) GetExecutionBlock(_ context.Context, id ids.ID) (validitywindow.ExecutionBlock[*ChunkCertificate], error) {
+	if blk, has := ti.blocks[id]; has {
+		return blk, nil
+	}
+	return nil, nil
```

Comment:
Fetching a non-existing block should return an error, can we return `database.ErrNotFound` if the block is not available?